docs/docs/dependency-specification.md

```diff
@@ -139,6 +139,24 @@ with the corresponding `add` call:
 poetry add https://example.com/my-package-0.1.0.tar.gz
 ```
 
+## `source` dependencies
+
+To depend on a package from an [alternate repository](/docs/repositories/#install-dependencies-from-a-private-repository),
+you can use the `source` property:
+
+```toml
+[[tool.poetry.source]]
+name = "foo"
+url = "https://foo.bar/simple/"
+secondary = true
+
+[tool.poetry.dependencies]
+my-cool-package = { version = "*", source = "foo" }
+```
+
```

Comment:
By the way, I just noticed this section doesn't mention the `add` command. So, similar to the `url` section just above this one, I would add something like:

````suggestion

with the corresponding `add` call:

```sh
poetry add my-cool-package --source foo
```

````